pkg/api/v1alpha1/cluster.go

```diff
@@ -250,18 +250,28 @@ func ParseClusterConfig(fileName string, clusterConfig KindAccessor) error {
 // from a multiobject yaml content. It doesn't set defaults nor validates the object
 func ParseClusterConfigFromContent(content []byte, clusterConfig KindAccessor) error {
 	for _, c := range strings.Split(string(content), YamlSeparator) {
-		if err := yaml.Unmarshal([]byte(c), clusterConfig); err != nil {
-			return err
-		}
-
-		if clusterConfig.Kind() == clusterConfig.ExpectedKind() {
-			return yaml.UnmarshalStrict([]byte(c), clusterConfig)
+		if kind := parseKind(c); kind != clusterConfig.ExpectedKind() {
+			continue
 		}
+		return yaml.UnmarshalStrict([]byte(c), clusterConfig)
 	}
 
 	return fmt.Errorf("yamlop content is invalid or does not contain kind %s", clusterConfig.ExpectedKind())
 }
 
+func parseKind(spec string) string {
+	kindRegex := regexp.MustCompile(`(?m)^kind: (.*)`)
```

Comment:
I would like to avoid a regex if possible

One possible solution would be to first unmarshall the object to a simple struct that just has a `Kind` string member. Example: https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/config_manager.go#L126
You can reset that object on each iteration so it doesn't carry any information from the previous one

Reply:
Done